hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -3388,7 +3387,8 @@ private CacheStoreMode determineCacheStoreMode(Map<String, Object> settings) {
 	}
 
 	private void checkTransactionNeeded() {
-		if ( !isTransactionInProgress() ) {
+		if ( !getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()
```

Comment:
I'm wondering about performance here, as this method is invoked frequently.

Specifically, the block:
 getSessionFactory().getSessionFactoryOptions().isAllowOutOfTransactionUpdateOperations()

Considering that this value will never change at runtime, what about making it a field of SessionImpl and initialize it during constructor?